vim-patch:8.0.{1384,1412,1414,1669,1702,1784},8.1.0060

janlazo opened the pull request.

janlazo committed:
vim-patch:8.0.1384: not enough quickfix help; confusing winid

Problem:    Not enough quickfix help; confusing winid.
Solution:   Add more examples in the help. When the quickfix window is not
            present, return zero for getqflist() with 'winid'. Add more tests
            for jumping to quickfix list entries. (Yegappan Lakshmanan, closes
            vim/vim#2427)
https://github.com/vim/vim/commit/74240d3febd1e3bc7cf086c647c9348b20716c33

janlazo committed:
vim-patch:8.0.1412: using free memory using setloclist()

Problem:    Using free memory using setloclist(). (Dominique Pelle)
Solution:   Mark location list context as still in use when needed. (Yegappan
            Lakshmanan, closes vim/vim#2462)
https://github.com/vim/vim/commit/12237448499aaeb8c4f2be7a1deda60c0f160627

janlazo committed:
vim-patch:8.0.1414: accessing freed memory in :lfile.

Problem:    Accessing freed memory in :lfile.
Solution:   Get the current window after executing autocommands. (Yegappan
            Lakshmanan, closes vim/vim#2473)
https://github.com/vim/vim/commit/14a4deb064610c30a50f00d524dde9b3292aad59

janlazo committed:
vim-patch:8.0.1669: :vimgrep may add entries to the wrong quickfix list

Problem:    :vimgrep may add entries to the wrong quickfix list.
Solution:   Use the list identifier. (Yegappan Lakshmanan)
https://github.com/vim/vim/commit/e1bb879f49665bb828197135b80aaf72cc190073

janlazo committed:
vim-patch:8.0.1784: gvim test gets stuck in dialog

Problem:    Gvim test gets stuck in dialog.
Solution:   Rename the file used.
https://github.com/vim/vim/commit/bc7845da935c0707e119812077cecd6cfb5a65e2

justinmk commented:
```
Direct leak of 240 byte(s) in 1 object(s) allocated from:
    #0 0x50ace8 in realloc (/home/travis/build/neovim/neovim/build/bin/nvim+0x50ace8)
    #1 0x112b6ec in xrealloc /home/travis/build/neovim/neovim/build/../src/nvim/memory.c:154:15
    #2 0xeb197a in ga_grow /home/travis/build/neovim/neovim/build/../src/nvim/garray.c:98:14
    #3 0x1477edd in addfile /home/travis/build/neovim/neovim/build/../src/nvim/path.c:1419:3
    #4 0x1486a2f in do_path_expand /home/travis/build/neovim/neovim/build/../src/nvim/path.c:715:13
    #5 0x147796b in path_expand /home/travis/build/neovim/neovim/build/../src/nvim/path.c:539:10
    #6 0x14755dd in gen_expand_wildcards /home/travis/build/neovim/neovim/build/../src/nvim/path.c:1243:32
    #7 0x147fb45 in expand_wildcards /home/travis/build/neovim/neovim/build/../src/nvim/path.c:2038:12
    #8 0xc31e24 in get_arglist_exp /home/travis/build/neovim/neovim/build/../src/nvim/ex_cmds2.c:1615:9
    #9 0x14ce726 in ex_vimgrep /home/travis/build/neovim/neovim/build/../src/nvim/quickfix.c:4101:7
```

janlazo committed:
vim-patch:8.1.0060: crash when autocommands delete the current buffer

Problem:    Crash when autocommands delete the current buffer. (Dominique
            Pelle)
Solution:   Check that autocommands don't change the buffer.
https://github.com/vim/vim/commit/600323b4ef51a58a8e800d8ca469383a3c911db7

janlazo committed:
vim-patch:8.0.1702: leaking memory when autocommands make quickfix list invalid

Problem:    Leaking memory when autocommands make a quickfix list invalid.
Solution:   Call FreeWild(). (Yegappan Lakshmanan)
https://github.com/vim/vim/commit/ee5b94a8ecbaeafa11c7c7c8cf7bef765fbea912
